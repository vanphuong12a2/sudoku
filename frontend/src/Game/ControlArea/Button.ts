export default class Button {
    public name: string;
    public onClickHandler: () => void;

    constructor(name: string, onClickHandler: () => void) {
        this.name = name;
        this.onClickHandler = onClickHandler;
    }
}