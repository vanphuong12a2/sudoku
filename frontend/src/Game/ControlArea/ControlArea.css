.control-area {
    margin: 1em;
}