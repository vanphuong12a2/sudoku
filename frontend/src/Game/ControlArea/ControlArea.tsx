import React from 'react';
import './ControlArea.css'
import Button from './Button';

const ControlFieldSet = (props: { legend: string, buttons: Button[] }) => {
    return (
        <fieldset>
            <legend>{props.legend}</legend>
            {
                props.buttons.map((button, buttonIndex) => {
                    return <ButtonComponent key={buttonIndex} button={button}/>
                })
            }
        </fieldset>
    );
};

const ButtonComponent = (props: { button: Button }) => {
    return (
        <a className='waves-effect waves-light btn-small' onClick={props.button.onClickHandler}>
            {props.button.name}
        </a>
    );
};

interface Props {
    newGameOnClickHandler: () => void
    clearBoardOnClickHandler: () => void
    refreshGameOnClickHandler: () => void
    solveGameOnClickHandler: () => void
}

const ControlArea = (props: Props) => {
    const gameButtons = [
        new Button('Refresh', props.refreshGameOnClickHandler),
        new Button('New game', props.newGameOnClickHandler)
    ];

    const solverButtons = [
        new Button('Clear all', props.clearBoardOnClickHandler),
        new Button('Show solution', props.solveGameOnClickHandler)
    ];
    return (
        <div className='control-area'>
            <ControlFieldSet legend='Game' buttons={gameButtons}/>
            <ControlFieldSet legend='Solver' buttons={solverButtons}/>
        </div>
    )
};

export default ControlArea;