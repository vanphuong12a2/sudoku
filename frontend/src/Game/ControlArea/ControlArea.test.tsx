import {create} from 'react-test-renderer';
import React from 'react';
import ControlArea from './ControlArea';

describe('<ControlArea />', () => {

    describe('Snapshots', () => {
        it('renders correctly', () => {
            const button = create(<ControlArea/>);
            expect(button.toJSON()).toMatchSnapshot();
        });
    });
});