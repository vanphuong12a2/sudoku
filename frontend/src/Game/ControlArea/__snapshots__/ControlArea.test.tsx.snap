// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlArea /> Snapshots renders correctly 1`] = `
<div
  className="control-area"
>
  <fieldset>
    <legend>
      Game
    </legend>
    <a
      className="waves-effect waves-light btn-small"
      onClick={[MockFunction]}
    >
      Refresh
    </a>
    <a
      className="waves-effect waves-light btn-small"
      onClick={[MockFunction]}
    >
      New game
    </a>
  </fieldset>
  <fieldset>
    <legend>
      Solver
    </legend>
    <a
      className="waves-effect waves-light btn-small"
      onClick={[MockFunction]}
    >
      Clear all
    </a>
    <a
      className="waves-effect waves-light btn-small"
      onClick={[MockFunction]}
    >
      Show solution
    </a>
  </fieldset>
</div>
`;
